src/ext/dogstatsd_client.c

```diff
@@ -0,0 +1,57 @@
+#include "dogstatsd_client.h"
+
+#include <dogstatsd_client/client.h>
+
+#include "configuration.h"
+#include "ddtrace.h"
+#include "logging.h"
+
+ZEND_EXTERN_MODULE_GLOBALS(ddtrace);
+
+#define METRICS_CONST_TAGS "lang:php,lang_version:" PHP_VERSION ",tracer_version:" PHP_DDTRACE_VERSION
+
+void ddtrace_dogstatsd_client_minit(TSRMLS_D) { DDTRACE_G(dogstatsd_client) = dogstatsd_client_default_ctor(); }
+
+static void _set_dogstatsd_client_globals(dogstatsd_client client, char *host, char *port, char *buffer TSRMLS_DC) {
+    DDTRACE_G(dogstatsd_client) = client;
+    DDTRACE_G(dogstatsd_host) = host;
+    DDTRACE_G(dogstatsd_port) = port;
+    DDTRACE_G(dogstatsd_buffer) = buffer;
+}
+
+void ddtrace_dogstatsd_client_rinit(TSRMLS_D) {
+    BOOL_T health_metrics_enabled = get_dd_trace_heath_metrics_enabled();
+    dogstatsd_client client = dogstatsd_client_default_ctor();
+    char *host = NULL;
+    char *port = NULL;
+    char *buffer = NULL;
+
+    if (health_metrics_enabled) {
+        host = get_dd_agent_host();
+        port = get_dd_dogstatsd_port();
+        buffer = malloc(DOGSTATSD_CLIENT_RECOMMENDED_MAX_MESSAGE_SIZE);
+        size_t len = DOGSTATSD_CLIENT_RECOMMENDED_MAX_MESSAGE_SIZE;
+
+        struct addrinfo *addrs;
+        int err;
+        if ((err = dogstatsd_client_getaddrinfo(&addrs, host, port))) {
```

Comment:
🔝 The error handling refactoring looks great!

Reply:
Thanks!